languages/rust/oso/src/oso.rs

```diff
@@ -13,7 +12,7 @@ use crate::ToPolar;
 
 #[derive(Clone)]
 pub struct Oso {
-    inner: Rc<polar_core::polar::Polar>,
+    inner: Arc<polar_core::polar::Polar>,
     host: Arc<Mutex<Host>>,
```

Comment:
Follow up: I don't think we need this to be shared between all queries - typically we clone the host and pass to the query. Would remove most of the contention in exchange for a more expensive clone up front.

Reply:
Have this noted down in my ticket.